src/applications/disability-benefits/2346/schemas/address-schema.js

```diff
@@ -241,9 +277,48 @@ export const addressUISchema = (
           },
         },
       },
+      province: {
+        'ui:title': 'State/Province/Region',
+        'ui:options': {
+          hideIf: (formData, index) => {
+            let countryNamePath = `${path}.country`;
+            if (typeof index === 'number') {
+              countryNamePath = insertArrayIndex(countryNamePath, index);
+            }
+            const livesOnMilitaryBase = get(livesOnMilitaryBasePath, formData);
+            if (isMilitaryBaseAddress && livesOnMilitaryBase) {
+              return true;
+            }
+            const countryName = get(countryNamePath, formData);
+            return countryName === USA.name || !countryName;
+          },
+        },
+        'ui:required': (formData, index) => {
+          if (
+            formData.selectedAddress === 'newAddress' &&
+            (formData.newAddress.country &&
+              formData.newAddress.country !== 'United States')
+          ) {
+            let countryNamePath = `${path}.country`;
+            if (typeof index === 'number') {
+              countryNamePath = insertArrayIndex(countryNamePath, index);
+            }
+            const livesOnMilitaryBase = get(livesOnMilitaryBasePath, formData);
+            if (isMilitaryBaseAddress && livesOnMilitaryBase) {
+              return true;
+            }
+            const countryName = get(countryNamePath, formData);
+            return countryName !== USA.name || !countryName;
+          }
+          return false;
+        },
+        'ui:errorMessages': {
+          required: 'Province is required',
```

Comment:
Maybe not important now, but this message should include a "Please" - see https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/platform/forms-system/src/js/definitions/address.js#L280